src/lekcja9_zadanie2_Samochody_klimatyzacja_spalanie/Truck.java

```diff
@@ -0,0 +1,48 @@
+package lekcja9_zadanie2_Samochody_klimatyzacja_spalanie;
+
+public class Truck extends Car {
+
+    private double loadWeight;
+
+    public double getLoadWeight() {
+        return loadWeight;
+    }
+
+    public void setLoadWeight(double loadWeight) {
+        this.loadWeight = loadWeight;
+    }
+
+    public Truck(String brand, String model, double tankCapacity, double averageFuelConsumptionPerHundredKm, boolean airConditioningOn, double loadWeight) {
+        super(brand, model, tankCapacity, averageFuelConsumptionPerHundredKm, airConditioningOn);
+        this.loadWeight = loadWeight;
+    }
+
+    @Override
+    public double getActualFuelConsumption() {
+        if (isAirConditioningOn() && getLoadWeight() > 0) {
+            double reslut = loadWeight * 0.5 / 100 + getAverageFuelConsumptionPerHundredKm() + 1.6;
+            return reslut;
+
+        } else if (!isAirConditioningOn() && getLoadWeight() > 0) {
+            double reslut = loadWeight * 0.5 / 100 + getAverageFuelConsumptionPerHundredKm();
+            return reslut;
+
+        } else if (isAirConditioningOn() && getLoadWeight() == 0) {
+            double reslut = getAverageFuelConsumptionPerHundredKm() + 1.6;
+            return reslut;
+
+        } else {
+            return getAverageFuelConsumptionPerHundredKm();
+        }
```

Comment:
to mozna troche uproscic. polecam popatrzec na rozwiazanie : http://git.javastart.pl/rozwiazania/samochody-i-spalanie-141077673/src/master/src/Truck.java

Reply:
Poprawione